packages/site/cms-site/vite.config.mts

```diff
@@ -0,0 +1,39 @@
+import react from "@vitejs/plugin-react";
+import preserveDirectives from "rollup-plugin-preserve-directives";
+import { defineConfig } from "vite";
+import dts from "vite-plugin-dts";
+import tsconfigPaths from "vite-tsconfig-paths";
+
+export default defineConfig({
+    plugins: [
+        tsconfigPaths({
+            projects: ["./tsconfig.build.json"],
+        }),
+        react(),
+        dts({ tsconfigPath: "./tsconfig.build.json" }), // generates the types for the library
+    ],
+    build: {
+        outDir: "lib",
+        lib: {
+            entry: "./src/index.ts",
+            fileName: () => "index.js",
+            formats: ["es"],
```

Comment:
when you only support here esmodules then you have to adapt the package.json to

Reply:
I updated package.json according to https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c in 9bdd969ede1d15a2fa4fc5be669c49eb289092b1. Is that okay?